olp-cpp-sdk-core/include/olp/core/client/OlpClientSettings.h

```diff
@@ -122,6 +127,18 @@ struct CORE_API AuthenticationSettings {
    */
   TokenProviderCallback provider;
 
+  /**
+   * @brief The user-provided function that returns ApiKey.
+   *
+   * If this provider is set it will be used instead of the token provider.
+   * The value returned from this, if not empty, will be added as URL parameter
+   * to each request.
+   *
+   * @note This method must to be sync and should not trigger any task on
+   * TaskScheduler as this might result in a deadlock.
+   */
```

Comment:
```suggestion
  /**
   * @brief The user-provided function that returns `ApiKey`.
   *
   * If this provider is set, it is used instead of the token provider.
   * The value returned from this, if not empty, is added as a URL parameter
   * to each request.
   *
   * @note This method must be synchronized and should not trigger any tasks on
   * `TaskScheduler` as it might result in a deadlock.
   */
```